hazelcast-sql/src/main/java/com/hazelcast/sql/impl/type/converter/Converter.java

```diff
@@ -369,4 +386,8 @@ private static boolean canConvert(Method method) {
     public String toString() {
         return getClass().getSimpleName();
     }
+
+    protected Object readResolve() throws ObjectStreamException {
```

Comment:
does `Converter` need to be serializable? It has many boolean fields but only `id` is sufficient in conjunction with `readResolve`

Reply:
`Converter` is serialized as part of `QueryDataType`.
1. We can mark other fields as `transient`.
2. We can specify custom Java serialization for `QueryDataType`, which only writes the ID of the converter, similar to `writeData`.

Since reusing `writeData` might be complicated or impossible in 2, I will adopt 1.

Reply:
Java serialization is very flexible in terms of backward compatibility, and I was already aware that our dependence on `Serializable` comes from lambdas. As Viliam pointed out, the suppliers, functions, etc. can be `IdentifiedDataSerializable`, but the complexity it brings doesn't usually justify the performance gain. (6f2d60eb)

Reply:
> Java serialization is very flexible in terms of backward compatibility,

flexible maybe, performant not that much. And in case of lambdas you do not control the names (and the algorithm for naming lambad and their fields AFAIR is not documented). It is easy to break compatibility by changing lambda to anonymous class.

But the explanation why Converter is Serializable is sufficient.